mail_tracking/static/src/js/failed_message.js

```diff
@@ -0,0 +1,384 @@
+                this._thread.fetchMessages(this.domain)
+                    .then(function () {
+                        var options = self._getThreadRenderingOptions();
+                        self._threadWidget.removeMessageAndRender(
+                            message.getID(), self._thread, options)
+                            .then(function () {
+                                self._updateButtonStatus(
+                                    !self._thread.hasMessages(), type);
+                            });
+                    });
+            } else {
+                this._super.apply(this, arguments);
+            }
+        },
+
+        _getThreadRenderingOptions: function () {
+            var values = this._super.apply(this, arguments);
+            if (this._thread.getID() === 'mailbox_failed') {
+                values.displayEmailIcons = true;
+                values.displayReplyIcons = false;
+                values.displayRetryButton = true;
+                values.displayReviewedButton = true;
+            }
+            return values;
+        },
+
+        _startListening: function () {
+            this._super.apply(this, arguments);
+            this.call('mail_service', 'getMailBus')
+                .on('update_failed', this, this._throttledUpdateThreads);
+        },
+
+        // Handlers
+        _onRetryFailedMessage: function (event) {
+            event.preventDefault();
+            var messageID = $(event.currentTarget).data('message-id');
+            this.do_action('mail.mail_resend_message_action', {
+                additional_context: {
+                    mail_message_to_resend: messageID,
+                },
+            });
+        },
+
+        _onMarkFailedMessageReviewed: function (event) {
+            event.preventDefault();
+            var messageID = $(event.currentTarget).data('message-id');
+            this._rpc({
+                model: 'mail.message',
+                method: 'toggle_tracking_status',
+                args: [[messageID]],
+                context: this.getSession().user_context,
+            });
+        },
+    });
+
+    MailManager.include({
+        _updateMailboxesFromServer: function (data) {
+            this._super.apply(this, arguments);
+            this._addMailbox({
+                id: 'failed',
+                name: _t("Failed"),
+                mailboxCounter: data.failed_counter || 0,
+            });
+        },
+    });
+
+    Mailbox.include({
+        _getThreadDomain: function () {
+            if (this._id === 'mailbox_failed') {
+                return [
+                    ['mail_tracking_ids.state', 'in', FAILED_STATES],
+                    ['mail_tracking_needs_action', '=', true],
+                ];
+            }
+            return this._super.apply(this, arguments);
+        },
+    });
+
+
+    /* FAILED MESSAGES CHATTER WIDGET */
+    function _readMessages (self, ids) {
+        if (!ids.length) {
+            return $.when([]);
```

Comment:
:thinking: ... what?

Reply:
?? Resolved promise if no ids given...
Similar to: https://github.com/odoo/odoo/blob/12.0/addons/mail/static/src/js/activity.js#L30

Reply:
Weird, I had seen this in the past:

```suggestion
            return $.when();
```

I guess it's mostly the same.